app/collections/collections-manager.jsx

```diff
@@ -1,46 +1,80 @@
 import React from 'react';
 import CollectionsCreateForm from './create-form';
 import CollectionSearch from './collection-search';
+import LoadingIndicator from '../components/loading-indicator';
 
 class CollectionsManager extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      adding: false,
       collections: [],
-      error: null,
-      hasCollectionSelected: false
+      errors: [],
+      hasCollectionSelected: true
     };
 
     this.addToCollections = this.addToCollections.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  onChange() {
+    const collections = this.search.getSelected();
+
+    if (collections.length > 0) {
+      this.setState({ hasCollectionSelected: false });
+    }
```

Comment:
Out of interest, if `collections` is an array of selected collections, why would `hasCollectionSelected` be false? I'm not sure I follow the logic here, which might be a problem for someone who comes along to maintain this later.